Configurations/15-android.conf

```diff
@@ -0,0 +1,207 @@
+#### Android...
+#
+# It takes *one* prior-set environment variable to make it work:
+#
+# ANDROID_NDK=/some/where/android-ndk-<ver>
+#
+# As well as PATH *adjusted* to cover ${CROSS_COMPILE}gcc and company.
+#
+# Note that it's different from original instructions that required to
+# set CROSS_SYSROOT [to $ANDROID_NDK/platforms/android-<api>/arch-<arch>]
+# and CROSS_COMPILE. CROSS_SYSROOT is still recognized [and even required
+# for some legacy targets], but if not set, it's detected and set to the
+# latest Android platform available with appointed NDK automatically. If
+# you need to target older platform, pass additional -D__ANDROID_API__=N
+# to Configure. For example, to compile for ICS on ARM with NDK 10d:
+#
+# ANDROID_NDK=/some/where/android-ndk-10d
+# PATH=$ANDROID_NDK/toolchains/arm-linux-androideabi-4.8/prebuild/linux-x86_64/bin:$PATH
+# [..]./Configure android-arm -D__ANDROID_API__=14
+#
+# One can engage clang by passing CC=clang to Configure. In such case
+# PATH needs even more adjustments to cover NDK's clang itself, as well
+# as unprefixed, yet target-specific ar and ranlib [or not, if you use
+# binutils-multiarch].
+
+{
+    my $android_ndk = {};
+    my %triplet = (
+        arm    => "arm-linux-androideabi",
+        arm64  => "aarch64-linux-android",
+        mips   => "mipsel-linux-android",
+        mips64 => "mips64el-linux-android",
+        x86    => "i686-linux-android",
+        x86_64 => "x86_64-linux-android",
+    );
+
+    sub android_ndk {
+        unless (%$android_ndk) {
+            my $ndk = $ENV{ANDROID_NDK};
+            die "\$ANDROID_NDK is not defined"  if (!$ndk);
+            die "\$ANDROID_NDK=$ndk is invalid" if (!-d "$ndk/platforms");
+
+            my $sysroot;
+
+            if (!($sysroot = $ENV{CROSS_SYSROOT})) {
+                my $api = "*";
+
+                # see if user passed -D__ANDROID_API__=N
+                foreach (@{$useradd{CPPDEFINES}}) {
+                    if (m|__ANDROID_API__=([0-9]+)|) {
+                        $api = $1;
+                        last;
+                    }
+                }
+
+                # list available platforms [numerically]
+                my @platforms = sort { $a =~ m/-([0-9]+)$/; my $aa = $1;
+                                       $b =~ m/-([0-9]+)$/; $aa <=> $1;
+                                     } glob("$ndk/platforms/android-$api");
+                die "no $ndk/platforms/android-$api" if ($#platforms < 0);
+
+                $config{target} =~ m|[^-]+-([^-]+)$|;   # split on dash
+                $sysroot = "@platforms[$#platforms]/arch-$1";
+            }
+            die "no sysroot=$sysroot"   if (!-d $sysroot);
+
+            $sysroot =~ m|/android-([0-9]+)/arch-(\w+)/?$|;
+            my ($api, $arch) = ($1, $2);
+
+            my $triarch = $triplet{$arch};
+            my $cflags = "-Wa,--noexecstack";
+            my $cppflags;
+
+            # see if user passed CC=clang
+            if ($user{CC} eq "clang") {
```

Comment:
Check for equality? It is common for builds  user to set CC plus some compiler options. In come cases CFLAGS and etc variables cannot be used. Is possible to check for substring clang in first word?

Reply:
Configure android-arm CC=clang -fmy-favorite-option -D__ARM_MAX_ARCH__=8 ...

Reply:
There is kind of contradiction in the statement. If you can't use CFLAGS and etc variables, then how come you can use CC, it's also a variable. What I'm saying is that find it hard time finding it compelling... Can you give an example? Specific to *this* project, i.e. what would be the problem you can't solve by passing extra flags. Also note that there is option for user to put together an extra .conf file that would meet overly specific needs...

Reply:
Thing about it is that I look explicitly for "clang" on $PATH in next line. So that *if* we choose to opt for a match instead of equality test [which I'm still not convinced about], it would have to be `|^clang\s|`. Well, of course one can alternatively choose to check for presence of clang on $PATH only when $user{CC} is specifically clang, and trust user to do the right thing otherwise... Well, there is fine line between hanging the user and giving him a rope...